src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSource.cs

```diff
@@ -17,20 +18,53 @@ public class PackageSource : IEquatable<PackageSource>
         /// </summary>
         public const int DefaultProtocolVersion = 2;
 
-        private readonly int _hashCode;
+        private int _hashCode;
+        private string _source;
+        private bool _isHttp;
+        private bool _isHttps;
+        private bool _isLocal;
+
+        public string Name { get; }
 
-        private bool? _isHttp;
-        private bool? _isHttps;
-        private bool? _isLocal;
+        public string Source
+        {
+            get => _source;
+            set
```

Comment:
Reading the issue description https://github.com/NuGet/Home/issues/10276, I guess, we should be good if we just make `Source` property read-only. Do we really need to go into this path of reevaluating all the properties based on a source change? `PackageSource` has `Credentials` property also which is tied to a single source. If we are going in this direction, I think we should also ensure that other related properties are also set with appropriate values.

Reply:
I started a conversation in Teams about exactly that idea, on the 4th of January, before I wrote tests and spent time making all the code prod-quality. Although the 4th of January was right after the New Year break, everyone was at work (well, except one who is on extended leave), so I think everyone had an opportunity to have a conversation. The only feedback I got at the time was that they thought it was overkill (although the idealist in me agrees with you, I wish this type was immutable). So, unfortunately there wasn't enough agreement in the team that spending a lot more effort to make API breaking changes was worth the benefit.